docs/proposals/scheduling/reschedule-task/reschedule-task.md

```diff
@@ -0,0 +1,336 @@
+    // +required
+    Kind string `json:"kind"`
+    
+    // Name of the target resource.
+    // +required
+    Name string `json:"name"`
+    
+    // Namespace of the target resource.
+    // Default is empty, which means it is a non-namespacescoped resource.
+    // +optional
+    Namespace string `json:"namespace,omitempty"`
+}
+
+// ScheduleTriggerStatus contains information about the current status of a ScheduleTrigger
+// updated periodically by schedule trigger controller.
+type ScheduleTriggerStatus struct {
+    // Phase represents the specific extent to which the task has been executed.
+    // Valid options are "Running", "Failed" and "Success", Defaults to "Running".
+    // +kubebuilder:default=Running
+    // +optional
+    Phase TriggerPhase `json:"phase,omitempty"`
+    
+    // FailedResourceList the list of trigger failed resources.
+    // +optional
+    FailedResourceList []FailedResource `json:"failedResourceList,omitempty"`
+}
+
+// TriggerPhase the specific extent to which the task has been executed
+type TriggerPhase string
+
+const (
+    // TriggerRunning the schedule trigger is on going, and hasn't finished its process.
+    TriggerRunning TriggerPhase = "Running"
+    // TriggerFailed some or all selected resource has been triggered a scheduling failed.
+    TriggerFailed TriggerPhase = "Failed"
+    // TriggerSuccess all selected resource has been triggered a scheduling success.
+    TriggerSuccess TriggerPhase = "Success"
+)
+
+// FailedResource trigger failed resource and its failure reason.
+type FailedResource struct {
+    TargetRefResource `json:",inline"`
+    
+    // FailReason the reason of resource triggered failed.
+    // +optional
+    FailReason string `json:"failReason,omitempty"`
+}
+
+// +kubebuilder:resource:scope="Cluster"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ScheduleTriggerList contains a list of ScheduleTrigger
+type ScheduleTriggerList struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ListMeta `json:"metadata,omitempty"`
+    
+    // Items holds a list of ScheduleTrigger.
+    Items []ScheduleTrigger `json:"items"`
+}
+```
+
+* Add two new fields to ResourceBinding/ClusterResourceBinding, one is in `spec` named `rescheduleTriggeredAt`, another
+is in `status` named `lastScheduledTime`, detail description is as follows:
+
+```go
+// ResourceBindingSpec represents the expectation of ResourceBinding.
+type ResourceBindingSpec struct {
+    ...
+	// RescheduleTriggeredAt is a timestamp representing when the referenced resource is triggered rescheduling.
+	// Only when this timestamp is later than timestamp in status.lastScheduledTime will the rescheduling actually execute.
+	//
+	// It is represented in RFC3339 form (like '2006-01-02T15:04:05Z') and is in UTC.
+	// It is recommended to be populated by the REST handler of command.karmada.io/Reschedule API.
+	// +optional
+	RescheduleTriggeredAt metav1.Time `json:"rescheduleTriggeredAt,omitempty"`
```

Comment:
1. It should be a pointer type for an optional field.
2. The comments should be explained in more detail, including what it is, value restrictions, and what will happen when setting.